### ISSUE TITLE
Q: check for REST API access (user impersonation or remote IP) in workflow

### ISSUE THREAD

**Issue by albrechtd** [anteo/redmine_custom_workflows#280]
First, this plugin is *really* great!

My problem: in a workflow, I would like to distinguish between a “real” browser session, and API access w/ user impersonation from a specific server.  I think this could be done
* either by the remote IP address of the request (as it will always come for a certain server),
* or by the information that an API access occurred (i.e. the access is done by the Admin user, but with `X-Redmine-Switch-User` impersonation).

Is it possible to get this information within the workflow?

**Comment by picman** [comment]
All those information are available in controllers. But we work with models in the plugin. I see no way how to get these information in the workflow script.

**Comment by albrechtd** [comment]
> All those information are available in controllers. But we work with models in the plugin. I see no way how to get these information in the workflow script.

That's really a pity – thanks a lot for you explanation anyway.  If you have *any* idea how I could solve my task, any input would be highly appreciated!

**Comment by picman** [comment]
The remote IP should be available now as follows. Try it, please, in the *devel* branch.
```ruby
self.custom_workflow_env[:remote_ip]
```

**Comment by albrechtd** [comment]
Thanks a lot for your changes, great work!

Unfortunately I forgot to mention that I'm running Redmine on a Debian Buster system, using the deb packages from [buster-backports](https://packages.debian.org/buster-backports/redmine) which are still v. 4.0.7, packages for Bullseye are not (yet) available, and I don't want to run Bookworm on a productive system.  Thus, I still use the latest plugin which is compatible with 4.0, which is v. 1.0.1…

However, I hacked v. 1.0.1 for issues (the part relevant for me) by adding [snippets](https://github.com/anteo/redmine_custom_workflows/files/9663621/v1.0.1-diff.zip) from your related commit, and this _seems_ to work perfectly.  I performed only a few tests, though, but they behave exactly as expected.
